docs/DeveloperGuide.md

```diff
@@ -1,38 +1,682 @@
+
+_Written by: Brian Wong Yun Long_
+
+### Purge Command
+
+The `PurgeCommand` inherits properties from the abstract `Command` class. The inheritance of `Command` from `PurgeCommand` is
+shown below.
+
+<p align="center">
+    <img src="images/PurgeCommandClassDiagram.png">
+    <br />
+    <i>Figure 3.5: Class Diagram for PurgeCommand Showing Inheritance of Command</i>
+</p>
+
+The full command for `purge` is `purge`.
+For example, if 'purge' is called, all transactions in Moolah Manager are removed.
+
+This is how the command works:
+
+1. The `main()` method in Duke calls `run()` in Duke. The `ui` reads the command via `ui.readCommand()` and parses it
+   through `CommandParser.parse()`.
+
+2. Within `CommandParser.parse()`, a few functions are called internally.
+   1. `spiltInput()` is called which splits the command from the parameter.
+   2. `getCommand()` is called which searches for the command.
+   3. `ParameterParser.parse()` is called.
+
+3. Within `ParameterParser.parse()`, a few functions are called internally as well.
+   1. `checkMandatoryTagsExist()` is called where the parameters are checked for all required tags exist based on the command.
+   2. `checkUnsupportedTagsNotExist()` is called to check if the parameter do not contain any unsupported tags based on the command.
+   3. `checkDuplicateTagsNotExist()` is called to check if the parameter do not contain any duplicate tags.
+   4. `checkParameterNotEmpty()` is called to check that the parameter inputted is not empty.
+   5. Once all these checks are successful, `setCommand()` is called.
+   
+4. Within `setCommand()`, there is no parameters required to be set for `purge`.
+
+5. The purge command is undergoing execution in `command.execute()` which will call functions within the PurgeCommand Class.
+   1. The function calls `isEmpty()` which returns `true` if the list of transactions is zero, `false` otherwise. It is stored 
+      in the local `check` variable.
+   2. The above function compares the size of the transactions list through the `transactions.size()` which is executed in
+      the TransactionList class and see if both are equal to zero.
+   3. The display will show an empty message if `isEmpty()` returns `true` via `ui.showInfoMessage()`, which exits the command.
+   4. Otherwise, a warning is displayed through `ui.showInfoMessage()` and reads in an input for the user to respond through `ui.readCommand()`.
+   5. If the input is `Y`, the command goes ahead and executes the `transactions.purgeTransactions()`. Any other input will
+      abort the command and the display will show an aborted message through `ui.showInfoMessage()`.
+   
+6. The `transactions.purgeTransactions()` function is executed in the TransactionList class.
+   1. The `transactions.clear()` function is called which deletes every single entry in Moolah Manager
+
+7. The display shows the successful purging via `ui.showInfoMessage()` and writes it to file by `storage.writeToFile()`.
```

Comment:
Same comment as above. I doubt developers would find this kind of description worth reading when they can read the code.